### PR TITLE
Added clickable/highlighted tables

### LINKED ISSUE #44: All tables should have clickable rows (see recent clients table for how to implement)

### PR DESCRIPTION
**Pull request by mhxu**
Also fixed a bug with clickable rows not being able to click on links/buttons in the row.  fixes #44

### CODE REVIEW

**Comment by hansonw** [comment]
Yeah, I'd add a rowClick attribute similar to the rowHref attribute to the _action_table template.  Cool i can edit your comment

**Comment by mhxu** [comment]
Hmm.  Our comments dissappeared.  I might of done something wrong.

File: app/views/templates/clients/_field_table.html.erb [hunk 1]
```diff
@@ -10,6 +10,7 @@
              :locals => {
                :name => 'data',
                :model => "client.#{field[:id]}.value",
+               :rowModule => "editInField('#{field[:id]}',$index)",
```
Comment (DouglasSherk) [inline app/views/templates/clients/_field_table.html.erb]: rowModal, though I'd actually prefer rowClick

File: app/views/templates/clients/_action_table.html.erb [hunk 2]
```diff
@@ -26,14 +26,15 @@
       <td colspan="<%= fields.length + 1 %>">No <%= name %> available.</td>
     </tr>
     <tr data-ng-repeat="item in <%= model %><%= ".slice(0, #{limit})" if !local_assigns[:limit].nil? %><%= " | orderBy:#{model}Predicate:#{model}Reverse" if local_assigns[:sortable] %>"
-        <%= defined?(rowhref) && raw("data-click-href='#{rowhref}'") %> >
+        <%= defined?(rowHref) && raw("data-click-href='#{rowHref}'") %>
+        <%= defined?(rowModule) && raw("data-ng-click=" + rowModule) %> >
```
Comment (DouglasSherk) [inline app/views/templates/clients/_action_table.html.erb]: Please add a TODO explaining that we should investigate this weird behavior.

File: app/assets/stylesheets/shared/_constants.css.scss [hunk 3]
```diff
@@ -9,4 +9,7 @@ $nav-width-desktops: 200px;
 $nav-width-widescreens: $nav-width-desktops;
 
 $light-gray: #f8f8f8;
-$light-green: #ddffdd;
\ No newline at end of file
+$light-green: #ddffdd;
+
+ // XXX: I dont know if this looks good or not
```
Comment (DouglasSherk) [inline app/assets/stylesheets/shared/_constants.css.scss]: Remove this comment.

File: app/assets/stylesheets/angular/clients.css.scss [hunk 4]
```diff
@@ -388,18 +393,15 @@ $client-form-nav-padding: 50px;
...
-
   tr th i {
     color: #666;
   }
+  
+  @include hover();
 }
 
 .history-table {
+
```
Comment (DouglasSherk) [inline app/assets/stylesheets/angular/clients.css.scss]: Unnecessary line break.

File: app/assets/stylesheets/shared/_utilities.css.scss [hunk 5]
```diff
@@ -55,3 +55,11 @@ $break-xlarge: 1200px;
   -moz-border-radius: $radius;
   border-radius: $radius;
 }
+
+@mixin hover() {
+  .ng-scope:hover {
```
Comment (DouglasSherk) [inline app/assets/stylesheets/shared/_utilities.css.scss:60]: If this only applies to ng-scope, we should make the mixin name more descriptive.

**Comment by mhxu** [comment]
I changed the stuff you asked for.  Should I create an issue for the TODO?

**Comment by DouglasSherk** [comment]
No, it's not important enough for an issue.

**Pull request merged**